contrib/buildsystems/CMakeLists.txt

```diff
@@ -35,25 +40,38 @@ empty(default) :
 NOTE: -DCMAKE_BUILD_TYPE is optional. For multi-config generators like Visual Studio
 this option is ignored
 
-This process generates a Makefile(Linux/*BSD/MacOS) , Visual Studio solution(Windows) by default.
+This process generates: a Makefile(Linux/*BSD/MacOS), Visual Studio solution(Windows) by default.
 Run `make` to build Git on Linux/*BSD/MacOS.
 Open git.sln on Windows and build Git.
 
 NOTE: By default CMake uses Makefile as the build tool on Linux and Visual Studio in Windows,
 to use another tool say `ninja` add this to the command line when configuring.
 `-G Ninja`
 
+The Visual Studio default generator changed in v16.6 from its visual studio
+implemenation to `Ninja` This required changes to many CMake scripts.
+
 ]]
 cmake_minimum_required(VERSION 3.14)
 
 #set the source directory to root of git
 set(CMAKE_SOURCE_DIR ${CMAKE_CURRENT_LIST_DIR}/../..)
 if(WIN32)
 	set(VCPKG_DIR "${CMAKE_SOURCE_DIR}/compat/vcbuild/vcpkg")
+	message("WIN32: ${WIN32}")
+	message("VCPKG_DIR: ${VCPKG_DIR}")
+	message("MSVC: ${MSVC}")
+	message("CMAKE_GENERATOR: ${CMAKE_GENERATOR}")
+	message("CMAKE_CXX_COMPILER_ID: ${CMAKE_CXX_COMPILER_ID}")
+	message("CMAKE_GENERATOR_PLATFORM: ${CMAKE_GENERATOR_PLATFORM}")
 	if(MSVC AND NOT EXISTS ${VCPKG_DIR})
 		message("Initializing vcpkg and building the Git's dependencies (this will take a while...)")
 		execute_process(COMMAND ${CMAKE_SOURCE_DIR}/compat/vcbuild/vcpkg_install.bat ${VCPKG_ARCH})
 	endif()
+	if(NOT EXISTS ${VCPKG_ARCH})
```

Comment:
It seems a bit weird to do this after we print out all the other build configuration information and run the vcpkg_install script using this option anyways, is there a reason it is printed out here rather than with the others?

Reply:
> a bit weird to do this after we print out all the other build configuration information and run the vcpkg_install script using this option anyways, is there a reason

The other printouts are as a reflection of the implicit Build Options. Dscho has asked that I include the `VCPKG_ARCH` in the list of those options.

The reason I put the message here is because the default arch is set within vcpkg_install script and I was keeping that dependency logic, so if a user ran vcpkg_install first, manually, without parameters, then we'd have the same end result.

Dscho made a similar comment wrt detecting the Visual Studio default, and the same logic should apply that we should only have one source of truth.

Hmm. Just rechecking. The on-line gui suggests you only highlighted one line, but maybe not. I'll need to check that the rebase onto upstream hasn't created some out of sequence interaction - I see an `AND MSVC` above which I wasn't expecting.

will re-check

Reply:
Yep. Looks like I goofed up the rebase / conflict. 

I'd also thought that the MSVC dependency had gone, because Visual Studio is now using Ninja, but maybe my memory is getting confused.

Reply:
Yeah, so I had a patch make it into seen that does do some work in regards to that dependency [here](https://github.com/gitgitgadget/git/pull/970).  Where we generate the compile_commands.json file by default.  So many tools can operate off of CMake builds with little work.

More relevant here we add the USE_VCPKG option, here so maybe it's better to base this off of that work as well?

Reply:
> 
> 
> Yeah, so I had a patch make it into seen that does do some work in regards to that dependency [here](https://github.com/gitgitgadget/git/pull/970). Where we generate the compile_commands.json file by default. So many tools can operate off of CMake builds with little work.
> 
> More relevant here we add the USE_VCPKG option, here so maybe it's better to base this off of that work as well?

Definitely. I had it in my mind that that work was already in GfW, and so would have been picked up when I rebased my series onto GfW (my upstream)...

I did a quick tweak of script to assume MSVC was true (by deleting it!), and VS has loaded vcpkg - Yay.

I'm preping to be away for a four-day weekend. Hopefully have something done on Wedn next.

Reply:
I lost track. Does this still need to be resolved, or has it been addressed already?

Reply:
I guess this concern has been addressed as part of a reply to [a different comment](https://github.com/git-for-windows/git/pull/3306#discussion_r706915084).